Fixes #24072 - freeip with compute resource via hostgroup

iNecas opened the pull request:
In #5790, we added an option to set the compute resource via hostgroup.
There was one thing missed though: the freeip not getting assigned
automagically, when the compute resource was selected this way.

After this change, we make sure to check for the ip addresses (and other
potential subnet related refreshes) after the hostgroup is selected, in
case the interfaces were populated during the hostgroup change.

iNecas committed:
Fixes #24072 - freeip with compute resource via hostgroup

In #5790, we added an option to set the compute resource via hostgroup.
There was one thing missed though: the freeip not getting assigned
automagically, when the compute resource was selected this way.

After this change, we make sure to check for the ip addresses (and other
potential subnet related refreshes) after the hostgroup is selected, in
case the interfaces were populated during the hostgroup change.

iNecas commented:
@xprazak2 since you worked on #5790, mind looking at this PR?

xprazak2 commented:
It does take care of ips not being loaded, however the compute attributes for network are not loaded when hostgroup with compute resource and compute profile is changed, so users have to open the network modal anyway to change them.

compute profile selected selected:
![](https://i.imgur.com/ds08kzC.png)

hostgroup selected:
![](https://i.imgur.com/rzoNqxM.png)

iNecas committed:
Refs #24072 - fix compute profile with CR set via HG

When compute resource is set via compute profile, we're not sending it
via the form (as it's disabled). Therefore, we need to search for in via
hostgroup.

iNecas committed:
Refs #24072 - fix selecting hostgroup with compute resource

Without this fix, the 'Virtual Machine' tab was giving:

   'mybox' not found on 'libvirt (Libvirt)' 'mybox' could
   be deleted or 'libvirt (Libvirt)' is not responding.

iNecas commented:
Thanks for pointing that out: it turned out there was more issues with the 'deploy on': I've extended the Redmine issue to mention the others as well (seems wasteful to create separate issues for those, as they all together mean the whole feature is not very usable).

I've added the fixes in separate commits, and I would prefer not squashing those, as it makes it easier to see what changes belong together, as opposed to one big commit.

I've also encouraged myself to write an integration tests for this. I've learned about the poltergeist more than I wished, but it helped finding some other issues as well and hopefully, it will help with regressions in this area in the future.

xprazak2 commented:
[test foreman]

xprazak2 commented:
Working nicely now, APJ.

iNecas commented on app/models/host/managed.rb:
```diff
@@ -559,8 +559,11 @@ def inherited_attributes
-    self.compute_attributes = compute_resource.compute_profile_attributes_for(compute_profile_id)
+    if compute_profile_present?
+      self.compute_attributes = compute_resource.compute_profile_attributes_for(compute_profile_id)
+    elsif compute_resource
```
Added check on compute resource presence when setting the compute attributes (with pure `else`, the bare-metal test was failing)

xprazak2 reviewed (approved):
:+1:

iNecas commented:
@ares mind having a look as someone with merge rights (if you know what  I mean ;-)
